extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -481,6 +507,104 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     }
   }
 
+  /**
+   * Sends RTC request as specified by rtcConfig. Returns promise which
+   * resolves to the targeting informtation from the request response.
+   * If the publisher has specified that noCache == true, then we will only
+   * send one RTC request per page, and it will always bypass cache. If
+   * noCache !== true, then we always send two requests. The first will try
+   * to hit browser cache, and the second request will always bypass.
+   * This is a way for us to implement a simple stale-while-revalidate
+   * model.
+   * @return {?Promise} The rtc targeting info to attach to the ad url.
+   * @private
+   */
+  sendRtcRequest_() {
+    // If we already sent the RTC request, return the value we've saved.
+    if (rtcPromise) {
+      return rtcPromise;
+    }
+
+    const ampRtcPageElement = document.getElementById('amp-rtc');
+    let rtcConfig;
```

Comment:
`if (!ampRtcPageElement) return Promise.resolve();`
`let rtcConfig;`
`let endPoint;`
`// NOTE usage of textContent instead of innerHTML`
`if (!isObject(rtcConfig = tryParseJson(ampRtcPageElement.textContent)) || !(endPoint = rtcConfig['endPoint']) || typeof endPoint != 'string' || !isSecureUrl(endPoint)) {`
   `user().warn(TAG, 'invalid RTC config...');`
   `return Promise.resolve();`
`}`